mergify_engine/queue.py

```diff
@@ -50,14 +55,55 @@ def log(self):
         )
 
     @classmethod
-    def from_context(cls, ctxt: context.Context) -> "Queue":
+    def from_context(cls, ctxt: context.Context, with_train: bool) -> "Queue":
+        train = (
+            merge_train.Train(
+                ctxt.client,
+                utils.get_redis_for_cache(),
+                ctxt.pull["base"]["repo"]["owner"]["id"],
+                ctxt.pull["base"]["repo"]["owner"]["login"],
+                ctxt.pull["base"]["repo"]["id"],
+                ctxt.pull["base"]["repo"]["name"],
+                ctxt.pull["base"]["ref"],
```

Comment:
Might be simpler to get a `github_types.GitHubRepository` as argument?

Reply:
I keep that for later.